modules/launcher/src/main/java/org/mule/module/launcher/domain/DefaultMuleDomain.java

```diff
@@ -318,10 +319,19 @@ public void dispose()
         {
             logger.info(miniSplash(String.format("Disposing domain '%s'", getArtifactName())));
         }
+
+        ClassLoader originalClassloader = Thread.currentThread().getContextClassLoader();
+
+        if (Thread.currentThread().getContextClassLoader() != deploymentClassLoader.getClassLoader())
```

Comment:
you should either remove this if here, or also ask if the classloader is different below when resetting the original classloader, for coinsistency